src/pcre2grep.c

```diff
@@ -3874,7 +3873,7 @@ else
   filename = name;
   }
 
-while ((patlen = sizeof(buffer)) && read_pattern(buffer, &patlen, f))
+while ((patlen = sizeof(buffer), read_pattern(buffer, &patlen, f)))
```

Comment:
Stupid coverity. It seems unable to reason about the `(patlen = sizeof(buffer)) &&` expression, which assigns to patlen, and then unconditionally continues (because `sizeof(buffer)` is always > 0).

Reply:
`-Wcomma` allows assignments on the LHS of a comma operator, so this should be accepted.

I'm not sure the original code _was_ clearer anyway. We're not logically doing an `&&`: we want to do something with side-effects on the left, ignore whatever it evaluates to, and then execute the right.

Reply:
I would rewrite it to `while (TRUE)`, set variable, break if read_pattern fails. This is not a nice code.

Reply:
Good point; I've done Zoltan's improvement.